src/test/java/com/opipo/codechallenge/service/TransactionAmountServiceTest.java

```diff
@@ -0,0 +1,112 @@
+        MockitoAnnotations.initMocks(this);
+    }
+
+    abstract protected T buildTransactionAmountService();
+
+    private static Stream<Arguments> provideAmounts() {
+        return Stream.of(Arguments.of(10d, 0.1d, 9.9d), Arguments.of(10d, 0d, 10d), Arguments.of(10d, 10d, 0d),
+                Arguments.of(10.1d, 0.1d, 10d), Arguments.of(-10d, 0.1d, -10.1d), Arguments.of(-4d, 8d, -12d),
+                Arguments.of(0d, 0.1d, -0.1d), Arguments.of(10d, null, 10d), Arguments.of(4d, 8d, -4d));
+    }
+
+    private static Stream<Arguments> provideTransactions() {
+        return Stream.of(
+                Arguments.of(-5d,
+                        new TransactionEntity[] {buildTransactionEntity(4d, 8d), buildTransactionEntity(15d, 16d)}),
+                Arguments.of(-43d,
+                        new TransactionEntity[] {buildTransactionEntity(-4d, 8d), buildTransactionEntity(-15d, 16d)}),
+                Arguments.of(-30d,
+                        new TransactionEntity[] {buildTransactionEntity(-10d, 2d), buildTransactionEntity(-15d, 3d)}),
+                Arguments.of(-35d,
+                        new TransactionEntity[] {buildTransactionEntity(-4d, null), buildTransactionEntity(-15d, 16d)}),
+                Arguments.of(45d,
+                        new TransactionEntity[] {buildTransactionEntity(10d, 0.1d), buildTransactionEntity(11d, 0.2d),
+                                buildTransactionEntity(12d, 0.3d), buildTransactionEntity(13d, 0.4d)}),
+                Arguments.of(4d, new TransactionEntity[] {buildTransactionEntity(23d, 19d)}));
+    }
+
+    private static Stream<Arguments> provideSums() {
+        return Stream.of(Arguments.of(10d, 1d, true), Arguments.of(0d, 10d, true), Arguments.of(-10d, 1d, false),
+                Arguments.of(1d, -10d, false), Arguments.of(null, -1d, false), Arguments.of(1d, null, true),
+                Arguments.of(1d, 1d, true), Arguments.of(1d, -1d, true));
+    }
+
+    private static TransactionEntity buildTransactionEntity(Double amount, Double fee) {
+        TransactionEntity transactionEntity = new TransactionEntity();
+        transactionEntity.setAmount(amount == null ? null : BigDecimal.valueOf(amount));
+        transactionEntity.setFee(fee == null ? null : BigDecimal.valueOf(fee));
+        return transactionEntity;
+    }
+
+    @ParameterizedTest(name = "Given amount {0} and fee {1}, {2} is expected")
+    @MethodSource("provideAmounts")
+    @DisplayName("Given amount and fee, then check the sum")
+    @Execution(ExecutionMode.CONCURRENT)
+    void calculateAmount(final Double amount, final Double fee, final Double result) {
+        assertEquals(BigDecimal.valueOf(result), transactionAmountService.calculateAmount(
+                amount == null ? null : BigDecimal.valueOf(amount), fee == null ? null : BigDecimal.valueOf(fee)));
+    }
+
+    @ParameterizedTest(name = "Given transaction with amount {0} and fee {1}, {2} is expected")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/albertinizao/codechallenge/pullRequest?prid=4681930)